### PR TITLE
Porting updates related to material datums.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
So, porting a lot of PRs from tgstation about material datums. This not includes everything that has been done so far. But heck I'm looking toward porting the rest later. credits to original authors, I may have missed some.

Also removed dwarves snowflake filth checking and gave em the snob trait instead.

## Changelog
:cl: Ghommie porting PRs by: nightred, defiantGrace, ArcaneMusic, Arkatos1, Firecage, Floyd / Qustinnus, uomo91, coiax
refactor: Changed custom material output on examine to be on a single line
add: makes toilets constructable
add: makes sinks constructable
add: Adds a new Lavaland Ruin, the Elephant Graveyard.
add: You can now learn to create several bone statues, and a new, dangerous shovel. The only thing stopping you are the consequences.
add: You can now appreciate rooms for being very well decorated.
tweak: Wood is now a datum material, and behaves like other datum materials.
/:cl:

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
Grave yard looks a bit off with the wires

File: code/modules/ruins/lavalandruin_code/elephantgraveyard.dm [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+		if(2)
+			new /obj/item/clothing/glasses/meson(src)
+		if(3)
+			new /obj/item/coin/silver(src)
+			new /obj/item/shovel/spade(src)
+		if(4)
+			new /obj/item/storage/book/bible/booze(src)
+		if(5)
+			new /obj/item/clothing/neck/stethoscope(src)
+			new	/obj/item/scalpel(src)
```
Comment (Trilbyspaceclone) [inline code/modules/ruins/lavalandruin_code/elephantgraveyard.dm:139]:
```suggestion
			new /obj/item/scalpel(src)
```

**Comment by Ghommie** [comment]
Everything nominal, I'm just using Kevinz cable network spawners because I'm lazy rn.

File: code/modules/ruins/lavalandruin_code/elephantgraveyard.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/ruins/lavalandruin_code/elephantgraveyard.dm:139]: Why?

Comment (Trilbyspaceclone) [reply code/modules/ruins/lavalandruin_code/elephantgraveyard.dm:139]: its off by one, not uniform

Comment (Trilbyspaceclone) [reply code/modules/ruins/lavalandruin_code/elephantgraveyard.dm:139]: Blep using tabs on this is hard do to needing to copy past, sorry for the mess up and 400 edits

Comment (Ghommie) [reply code/modules/ruins/lavalandruin_code/elephantgraveyard.dm:139]: Oh, I misread the suggestion, thanks I guess.

**Review by silicons — COMMENTED** [review]
is beauty a good idea with our memory usage limitations right now?

**Comment by Ghommie** [comment]
Made the art and beauty components into elements, are you fine with it now?

**Pull request merged**